cdc/log.cc

```diff
@@ -591,15 +697,21 @@ class transformer final {
             }
         } else {
             // should be insert, update or deletion
+            std::optional<clustering_key> poikey;
             auto process_cells = [&](const row& r, column_kind ckind, const clustering_key& log_ck, std::optional<clustering_key> pikey, const cql3::untyped_result_set_row* pirow) -> std::optional<gc_clock::duration> {
-                bytes_opt value;
+                if (postimage && !poikey) {
+                    poikey = set_pk_columns(m.key(), ts, tuuid, ++batch_no, res);
+                    set_operation(*poikey, ts, operation::post_image, res);
+                }
```

Comment:
Why are we setting `poikey` here and not near the place where `pikey` and `log_ck` are set? This choice of placement of this code seems pretty random to me.

Reply:
To avoid code duplication. Since we only either do static cols or regular cols, this is fine. There will be a separate transform call for each.